src/contentScript.js

```diff
@@ -1,3 +1,79 @@
+chrome.runtime.onMessage.addListener(
+	function (request, sender, sendResponse) {
+		console.log(sender.tab ?
+			"from a content script:" + sender.tab.url :
+			"from the extension");
+
+		// alert(request.message);
+		if (!!request.dont_show_again_preference && request.dont_show_again_preference){ 
```

Comment:
What is this?

Reply:
a pop is shown to a user when they submit an infraction
this is used as preference controls so the user may choose to dismiss the popup and never show it again or just dismiss it and it will be shown the next time that they submit an infraction

Reply:
I meant the ‘!!’ Operator